packages/core/contracts/financial-templates/common/FundingRatePayer.sol

```diff
@@ -0,0 +1,121 @@
+ * to access a funding rate. Contract is abstract as each derived contract that inherits `FundingRatePayer` must
+ * implement `pfc()`.
+ */
+
+abstract contract FundingRatePayer is FeePayer {
+    /****************************************
+     *      FEE PAYER DATA STRUCTURES       *
+     ****************************************/
+
+    // Tracks the last block time when the fees were paid.
+    uint256 private lastPaymentTime;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event FundingRateFeesPaid(uint256 indexed fundingRateFee, uint256 indexed lateFee);
+
+    /****************************************
+     *              MODIFIERS               *
+     ****************************************/
+
+    // modifier that calls continuous fee paying methods.
+    // TODO: Prettier doesn't allow modifiers to be overridable?
+    // prettier-ignore
+    modifier fees override {
+        payRegularFees();
+        payFundingRateFees();
+        _;
+    }
+
+    /**
+     * @notice Constructs the FundingRatePayer contract. Called by child contracts.
+     * @param _collateralAddress ERC20 token that is used as the underlying collateral for the synthetic.
+     * @param _finderAddress UMA protocol Finder used to discover other protocol contracts.
+     * @param _timerAddress Contract that stores the current time in a testing environment.
+     * Must be set to 0x0 for production environments that use live time.
+     */
+    constructor(
+        address _collateralAddress,
+        address _finderAddress,
+        address _timerAddress
+    ) public FeePayer(_collateralAddress, _finderAddress, _timerAddress) {
+        lastPaymentTime = getCurrentTime();
+    }
+
+    /****************************************
+     *        FEE PAYMENT FUNCTIONS         *
+     ****************************************/
+
+    /**
+     * @notice Pays continuous fees (as a % of the collateral pool) to the FundingRateStore contract.
+     * @dev These must be paid periodically for the life of the contract. If the contract has not paid its fee
+     * in a week or more then a late penalty is applied which is sent to the caller. If the amount of
+     * fees owed are greater than the pfc, then this will pay as much as possible from the available collateral.
+     * An event is only fired if the fees charged are greater than 0.
+     * @return totalPaid Amount of collateral that the contract paid (sum of the amount paid to the Store and caller).
+     * This returns 0 and exit early if there is no pfc, fees were already paid during the current block, or the fee rate is 0.
+     */
+    function payFundingRateFees() public nonReentrant() returns (FixedPoint.Unsigned memory totalPaid) {
+        FundingRateStoreInterface fundingRateStore = FundingRateStoreInterface(
```

Comment:
This is mostly copied logic from `payRegularFees`, so I should consider refactoring shared logic

Reply:
Yeah.

Just out of curiosity, I wonder if it would be simpler just to fork the FeePayer and have it do two payments instead of 1. Not sure if that would be simpler though.

What are your thoughts?

Reply:
I don't think that would simpler, I actually benefit from using the internal cumulative fee multiplier modifying functions and its cleaner to just extend the `fees()` modifier instead of adding another one